### ISSUE TITLE
Filtering by list property length

### ISSUE THREAD

**Issue by kevboh** [realm/realm-js#1139]
## Goals

I'd like to filter a parent object by the number of child objects in a list property. For example, if I have a `Person` who has a list of `Car`s, I'd like to find everyone with no cars, something like:

```
realm.objects('Person').filtered('cars.length == 0');
```

## Expected Results

I expected this to return all `Person` objects with a `cars` list of length 0.

## Actual Results

```
Error: No property 'length' on object of type 'Person'
```

## Version of Realm and Tooling

- Realm JS SDK Version: 1.9.0
- Node or React Native: both (node scripts, RN app)
- Client OS & Version: Node v7.9.0, RN 0.45.1
- Which debugger for React Native: None

**Comment by kristiandupont** [comment]
This would be useful, yes. We'll keep it in mind.

**Comment by ironage** [comment]
This is implemented since version 2.2.0.
You can use `@size` or `@count` interchangeably on lists, string properties or binary properties.
In the example above it would be:
`realm.objects('Person').filtered('cars.@count == 0');`
